reconcile/ocm_machine_pools.py

```diff
@@ -123,14 +156,27 @@ class NodePool(AbstractPool):
     aws_node_pool: AWSNodePool
     subnet: Optional[str]
 
+    @staticmethod
+    def _patch_autoscaling(spec: Mapping[str, Any]):
```

Comment:
Instead of always patch it, can we do conditional patch for hypershift cluster only. So this can support machine pool operations in both cases. Why `hypershift` is not using the same `MachinePoolAutoscaling` schema in https://api.openshift.com/?urls.primaryName=Clusters%20management%20service#/default/post_api_clusters_mgmt_v1_clusters__cluster_id__machine_pools